### ISSUE TITLE
[Restore] Layer metadata restorer should not write layer.sha files for platform API >= 0.7

### ISSUE THREAD

**Issue by natalieparellano** [buildpacks/lifecycle#584]
### Description
For earlier platform API, the analyzer would write `<layers>/<buidpack-id>/<layer>.sha` files containing the diffID of each analyzed layer. These files are used by the restorer to verify that the layer metadata written by the analyzer was for the same image / layer that the restorer has access to.

Now that metadata and cache restoration are happening in the same phase in platform API >= 0.7, these files are no longer needed. 

### Proposed solution
Layer metadata restorer should not write layer.sha files for platform API >= 0.7

### Describe alternatives you've considered
This could be folded into https://github.com/buildpacks/lifecycle/pull/582, but we'd like to make this a separate unit of work in order to unblock other issues that rely on https://github.com/buildpacks/lifecycle/pull/582.

### Additional context
Spec PR: https://github.com/buildpacks/spec/pull/197

**Comment by natalieparellano** [comment]
Blocked on https://github.com/buildpacks/lifecycle/pull/582

**Comment by natalieparellano** [comment]
cc @jabrown85
